aptos-move/block-executor/src/executor.rs

```diff
@@ -1546,7 +1549,27 @@ where
                     executor_arguments,
                     signature_verified_block,
                     base_view,
-                    // Always enable resource group serialization handling in fallback.
+                    resource_group_bcs_fallback,
+                );
+            }
+
+            if !resource_group_bcs_fallback
```

Comment:
how about making the first iff above be && !resource_group_bcs_fallback, and pass false to execute_transactions_sequential

and then have this if be top-level - the only if that does resource_group_bcs_fallback? (and you need to recompute resource_group_bcs_fallback here, based on the new ret

Reply:
yes, much better